### ISSUE TITLE
Wrecks pre-existing Atom installs

### ISSUE THREAD

**Issue by dhinklexo** [luna/luna-studio#705]
Installed the latest Luna Studio beta on my machine which has an existing Atom setup. Now my normal Atom setup:

1. Has invalid plugins
2. Was reverted to an earlier version
3. Now launches with the "Luna Studio" icon
4. Getting core Atom issues

Now a quick glance through the the beta has wrecked my development environment which is extremely unfortunate. As a quick fix, it would be nice if the installer at least threw an error when Atom is pre-installed.

**Comment by piotrMocz** [comment]
May I ask what system you have? And which version? May it be possible that you are actually running the Atom from inside the package and not your regular one?
We went to great lengths to ensure separation of the bundled Atom from any other installs on the system, so it's very sad to hear about any issues.

Anyways, please let us know your setup (ideally post some logs or screenshots) and we will get to debugging :)

**Comment by dhinklexo** [comment]
MacOS Sierra 10.12.6, originally running latest Atom.

I launch Atom via the terminal command 'atom'. After installing Luna Studio, this caused Atom to launch with my correct theme and plugins, but with several additional plugins, and it was an older version of Atom launching. 

I removed Luna studio, ran Atom's update, and it seems to no longer have the additional plugins or related errors, though it complained about a missing config on first load (which didn't seem to affect anything). 

Is Luna studio possibly introducing a config to `~/.atom` directly or via symlink? Or is this maybe a bug with Atom caching configs incorrectly?

**Comment by z5h** [comment]
MacOS 10.13.2 (17C205)
I had a similar experience to @dhinklexo 

I removed `.luna` and the `Luna Studio.app` and the line it put in my `.zshrc` which incidentally, made it so I couldn't see part of my path. I have yet to understand/diagnose that.

**Comment by piotrMocz** [comment]
I would be leaning towards suspecting that there's nothing wrong with the "original" Atom. Much more likely is that something added to `PATH` causes the `atom` command to point to the Atom from inside `Luna Studio.app`.

We do not write anything to `.atom` or any of the system-wide config folders, we fixed that a while ago (it was especially funny on some linuxes).

We need to stop prepending stuff to user's `PATH`, though, it's one of our tasks (we may prompt about that, but not do it by default).

**Comment by piotrMocz** [comment]
One more thing to check would be the `ATOM_HOME` environment variable. Could you check, before running `atom`, whether that's set to `~/.atom`, or some Luna folder? It should not happen, but maybe the variable was left set to a value that we use.

**Comment by dhinklexo** [comment]
I reverted it on my main dev machine, but I will see if I can repro on a spare macbook (which has a similar setup) tonight and do some analysis.

**Comment by somlor** [comment]
FYI I read [another report here](https://lobste.rs/s/tao183/luna_hybrid_visual_textual_functional#c_ootdvw) of same issue. As an Atom user with very customized setup I'm hesitant to install until knowing this is resolved. Thank you, excited to try Luna!

**Comment by dhinklexo** [comment]
@piotrMocz Ok, looks like your theory about the PATH being modified is correct. I repro'd on my other machine and after installing luna this is what my path looks like:
`PATH=/Users/kyros/.local/bin/luna-studio:/usr/local/bin:/usr/local/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/kyros/npm/bin`

Running `atom .` loads the Luna atom instance (though it doesn't appear correctly? That's a different issue) since it has the first PATH entry.

**Comment by rpowelll** [comment]
@somlor oh hey that's me. I looked into this a little further and it seems that Luna replaced my `atom` executable at `/usr/local/atom` with one that launched it's own instance. My hunch is that this is some default Atom behaviour that they didn't disable in their fork.

**Comment by piotrMocz** [comment]
@rpowelll we will take a look at this immediately -- this is something we would like to avoid like a plague. Are you sure the executable is replaced, and not just overshadowed like @dhinklexo mentioned?

@dhinklexo running the bundled Atom directly (as opposed to running the `LunaStudio.app`) will produce poor results, as it already has the plugins installed and they need some additional env vars to run correctly -- this is expected.

**Comment by dhinklexo** [comment]
@piotrMocz Right, the issue is that running `atom` should in fact, run MY pre-installed atom. Luna shouldn't (hopefully) be messing with my path *at all*, but probably symlink it's own executable (named something else) to a proper directory. 

`atom` is a correct, valid, and de facto way of launching the atom text editor. Luna shouldn't be hijacking it.

**Comment by piotrMocz** [comment]
@dhinklexo I agree, this is bad and we will try to fix it asap.

By the way, if you'd like to see the way Luna is installed in more detail, you could check out [the source for the install command](https://github.com/luna/luna-manager/blob/master/luna-manager/src/Luna/Manager/Command/Install.hs) ;)

**Comment by rpowelll** [comment]
@piotrMocz yeah I ran `which atom` to see if something with higher precedence in my `PATH` was shadowing Atom's version of the executable, but that command printed `/usr/local/bin/atom`

**Comment by sylwiabr** [comment]
does it still happens with new packages and newest installer? I can't reproduce it on my macos.

**Comment by somlor** [comment]
Been waiting to see this issue resolved before installing Luna Studio, curious re: current status?
